module/library/categories/helper.js

```diff
@@ -321,9 +394,65 @@ module.exports = class LibraryCategoriesHelper {
 	 * @returns {Object} category details
 	 */
 
-	static create(categoryData) {
+	static create(categoryData, files) {
 		return new Promise(async (resolve, reject) => {
 			try {
+				if (files && files.cover_image) {
+					let coverImages = files.cover_image
+
+					// Generate a unique ID for the file upload
+					let uniqueId = await UTILS.generateUniqueId()
+
+					// Prepare the request data for the file upload
+					let requestData = {
+						[uniqueId]: {
+							files: [],
+						},
+					}
+
+					for (let file of coverImages) {
+						requestData[uniqueId].files.push(file.name)
+					}
+
+					let signedUrl = await filesHelpers.preSignedUrls(requestData, '', false, 'cover_image')
+
+					if (
+						signedUrl.data &&
+						Object.keys(signedUrl.data).length > 0 &&
+						signedUrl.data[uniqueId].files.length > 0 &&
+						signedUrl.data[uniqueId].files[0].url &&
+						signedUrl.data[uniqueId].files[0].url !== ''
+					) {
+						for (let fileFromRequest of coverImages) {
+							let fileUploadUrl = signedUrl.data[uniqueId].files.filter((fileData) => {
+								return fileData.file == fileFromRequest.name
+							})
+
+							const uploadData = await axios.put(fileUploadUrl[0].url, fileFromRequest.data, {
+								headers: {
+									'x-ms-blob-type':
+										process.env.CLOUD_STORAGE_PROVIDER === 'azure' ? 'BlockBlob' : null,
+									'Content-Type': 'multipart/form-data',
+								},
+							})
+
+							if (!(uploadData.status == 200 || uploadData.status == 201)) {
+								throw new Error('File upload failed.')
+							}
+						}
+					}
+
+					let i = 0
```

Comment:
@borkarsaish65 instead of i call it sequenceNumber

Reply:
done